tests/energy-profiler/test_csv_export.cpp

```diff
@@ -0,0 +1,72 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <chrono>
+#include <thread>
+#include <string>
+#include "gtest/gtest.h"
+#include "../../profiling/energy-profiler/common/timer_system.hpp"
+
+using namespace KokkosTools::EnergyProfiler;
+
+/**
+ * @test This test checks that the CSV export functions work correctly
+ */
+TEST(EnergyProfilerCSVTest, CSVExport) {
+  KernelTimerTool timer;
+
+  // Simulate some kernel operations
+  timer.start_region("test_kernel_1", RegionType::ParallelFor, 1);
+  std::this_thread::sleep_for(std::chrono::milliseconds(10));
+  timer.end_region();
+
+  timer.start_region("test_kernel_2", RegionType::ParallelReduce, 2);
+  std::this_thread::sleep_for(std::chrono::milliseconds(5));
+  timer.end_region();
+
+  timer.start_region("test_region", RegionType::UserRegion, 3);
+  std::this_thread::sleep_for(std::chrono::milliseconds(15));
+  timer.end_region();
+
+  timer.start_region("test_deepcopy", RegionType::DeepCopy, 4);
+  std::this_thread::sleep_for(std::chrono::milliseconds(8));
+  timer.end_region();
+
+  // Test the CSV export functions
+  const auto& kernels    = timer.get_kernel_timings();
+  const auto& regions    = timer.get_region_timings();
+  const auto& deepcopies = timer.get_deep_copy_timings();
+
+  // Expected: 2 kernels (ParallelFor and ParallelReduce), 1 region, 1 deep copy
+  ASSERT_EQ(kernels.size(), 2)
+      << "Expected 2 kernels, found " << kernels.size();
```

Comment:
You don't need to add this extra information, if it fails GTest will say kernels.sizeIO was not 2 and print the actual value.